src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -0,0 +1,107 @@
+{
+    /// <summary>
+    /// <see cref="ITagHelper"/> implementation targeting &lt;link&gt; elements that supports fallback href paths.
+    /// </summary>
+    public class LinkTagHelper : TagHelper
+    {
+        private const string FallbackHrefAttributeName = "asp-fallback-href";
+        private const string FallbackTestClassAttributeName = "asp-fallback-test-class";
+        private const string FallbackTestPropertyAttributeName = "asp-fallback-test-property";
+        private const string FallbackTestValueAttributeName = "asp-fallback-test-value";
+        private const string FallbackTestMetaTemplate = "<meta name=\"x-stylesheet-fallback-test\" class=\"{0}\" />";
+        private const string FallbackJavaScriptResourceName = "compiler/resources/LinkTagHelper_FallbackJavaScript.js";
+
+        // NOTE: All attributes are required for the LinkTagHelper to process.
+        private static readonly string[] RequiredAttributes = new[]
+        {
+            FallbackHrefAttributeName,
+            FallbackTestClassAttributeName,
+            FallbackTestPropertyAttributeName,
+            FallbackTestValueAttributeName
+        };
+
+        /// <summary>
+        /// The URL of a CSS stylesheet to fallback to in the case the primary one fails (as specified in the href
+        /// attribute).
+        /// </summary>
+        [HtmlAttributeName(FallbackHrefAttributeName)]
+        public string FallbackHref { get; set; }
+
+        /// <summary>
+        /// The class name defined in the stylesheet to use for the fallback test.
+        /// </summary>
+        [HtmlAttributeName(FallbackTestClassAttributeName)]
+        public string FallbackTestClass { get; set; }
+
+        /// <summary>
+        /// The CSS property name to use for the fallback test.
+        /// </summary>
+        [HtmlAttributeName(FallbackTestPropertyAttributeName)]
+        public string FallbackTestProperty { get; set; }
+
+        /// <summary>
+        /// The CSS property value to use for the fallback test.
+        /// </summary>
+        [HtmlAttributeName(FallbackTestValueAttributeName)]
+        public string FallbackTestValue { get; set; }
+
+        // Protected to ensure subclasses are correctly activated. Internal for ease of use when testing.
+        // TODO: Change this to activate ILogger<LinkTagHelper> directly once https://github.com/aspnet/Hosting/pull/147 is in
+        [Activate]
+        protected internal ILoggerFactory LoggerFactory { get; set; }
+
+        /// <inheritdoc />
+        public override void Process(TagHelperContext context, TagHelperOutput output)
+        {
+            var logger = LoggerFactory.Create<LinkTagHelper>();
+
+            if (!context.ShouldProcess(RequiredAttributes, logger))
+            {
+                // TODO: Should we trace the current view/request path here too?
+                logger.WriteVerbose("Skipping processing for {0} {1}", nameof(LinkTagHelper), context.UniqueId);
```

Comment:
Not sure if it's common practice to provide the reason for why we're logging information but I can see value in adding "why" processing was skipped.